rust/loqui_client/src/connection_handler.rs

```diff
@@ -171,12 +176,17 @@ impl<E: Encoder> ConnectionHandler<E> {
         payload: E::Encoded,
         encoding: &'static str,
         sequence_id: u32,
-        waiter_tx: OneShotSender<Result<E::Decoded, Error>>,
+        waiter: ResponseWaiter<E::Decoded>,
     ) -> Option<LoquiFrame> {
+        if waiter.deadline < Instant::now() {
```

Comment:
I know this is super rare, but we should probably make comparing either be `>=` and `<` or `<=` and `>`, but not `>` and `<`, which it to say:

We compare above:
```rust
self.waiters.retain(|_sequence_id, waiter| waiter.deadline > now);
```

And here:
```
if waiter.deadline < Instant::now() {
```

This leave the (super rare - probably will never happen) state where `waiter.deadline == Instant::now()` un-handled.